workloads/virtual-machines/disks-and-volumes.md

```diff
@@ -373,3 +375,120 @@ spec:
 
 Ephemeral VMs very often come with read-only root images and limited tmpfs space. In many cases this is not enough to install application dependencies and provide enough disk space for the application data. While this data is not critical and thus can be lost, it is still needed for the application to function properly during its lifetime. This is where an `emptyDisk` can be useful. An emptyDisk is often used and mounted somewhere in `/var/lib` or `/var/run`.
 
+### configMap
+
+A `configMap` is a reference to a [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) in Kubernetes. An extra `iso` disk will be allocated which has to be mounted on a VM. To mount the `configMap` users can use a `cloudInit` and disk serial number. The `configMapName` needs to be specified for a reference to the created kubernetes `ConfigMap` with the same name.
+
+> **Note:** Currently, the ConfigMap update propagation is not supported. After the update, only a pod will be aware of changes, not running VMI.
+
+Example: Attach the `configMap` to a VM and use a `cloudInit` to mount the `iso` disk:
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VirtualMachineInstance
+metadata:
+  creationTimestamp: null
+  labels:
+    special: vmi-fedora
+  name: vmi-fedora
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: registrydisk
+        volumeName: registryvolume
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+        volumeName: cloudinitvolume
+      - disk: {}
+        name: app-config-disk
+        volumeName: configmap-volume
+        # set serial
+        serial: CVLY623300HK240D
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  terminationGracePeriodSeconds: 0
+  volumes:
+  - name: registryvolume
+    registryDisk:
+      image: registry:5000/kubevirt/fedora-cloud-registry-disk-demo:devel
+  - cloudInitNoCloud:
+      userData: |-
+        #cloud-config
+        password: fedora
+        chpasswd: { expire: False }
+        bootcmd:
+          # mount the ConfigMap
+          - "mkdir /mnt/app-config"
+          - "mount /dev/$(lsblk --nodeps -no name,serial | grep CVLY623300HK240D | cut -f1 -d' ') /mnt/app-config"
+    name: cloudinitvolume
+  - configMap:
+      configMapName: app-config
```

Comment:
I think that the examples need to be adjusted to align with the latest code changes. @mfranczy can that be?

Reply:
Done. I added a note about ```items``` as well.